File: SYP.Web/Modules/Catalog/PriceLists/Editor/PriceListsEditorColumns.cs
namespace SYP.Catalog.Columns;

[ColumnsScript("Catalog.PriceListsEditor")]
[BasedOnRow(typeof(PriceListsRow), CheckNames = true)]
public class PriceListsEditorColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Code { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public string CurrencyCode { get; set; }
    public DateTime ValidFrom { get; set; }
    public DateTime ValidTo { get; set; }
    public short IsActive { get; set; }
    public bool IsDefault { get; set; }
    public DateTime InsertDate { get; set; }
    public int InsertUserId { get; set; }
    public DateTime UpdateDate { get; set; }
    public int UpdateUserId { get; set; }
}

File: SYP.Web/Modules/ServerTypes/Catalog/PriceListsEditorColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { PriceListsRow } from "./PriceListsRow";

export interface PriceListsEditorColumns {
    Id: Column<PriceListsRow>;
    Code: Column<PriceListsRow>;
    Name: Column<PriceListsRow>;
    Description: Column<PriceListsRow>;
    CurrencyCode: Column<PriceListsRow>;
    ValidFrom: Column<PriceListsRow>;
    ValidTo: Column<PriceListsRow>;
    IsActive: Column<PriceListsRow>;
    IsDefault: Column<PriceListsRow>;
    InsertDate: Column<PriceListsRow>;
    InsertUserId: Column<PriceListsRow>;
    UpdateDate: Column<PriceListsRow>;
    UpdateUserId: Column<PriceListsRow>;
}

export class PriceListsEditorColumns extends ColumnsBase<PriceListsRow> {
    static readonly columnsKey = 'Catalog.PriceListsEditor';
    static readonly Fields = fieldsProxy<PriceListsEditorColumns>();
}